src/TitleReceive/TitleReceiveList.js

```diff
@@ -78,13 +83,19 @@ export const TitleReceiveList = ({
             fullWidth
           />
         ),
-        location: record => (
-          <FieldLocationFinal
-            locationId={fields.value[record.rowIndex]?.locationId}
-            onChange={({ id }) => selectLocation(id, `${field}[${record.rowIndex}].locationId`)}
-            name={`${field}[${record.rowIndex}].locationId`}
-          />
-        ),
+        location: record => {
+          const locationId = fields.value[record.rowIndex]?.locationId;
+          const locationIds = locationId ? uniqBy([...poLineLocationIds, locationId]) : poLineLocationIds;
```

Comment:
1. IMHO, such operations could be done through the `Set`, and in one place instead of every call of common code.
2. What is about API requests? if there are 11 pieces in Receiving screen for example.

Reply:
For all pieces pol locations are the same, but each piece can have a unique location from plugin

Reply:
so it's like 11 equal requests if every piece has the same location?

Reply:
what if there are 50 pieces?

Reply:
Updated logic for fetching locations